src/core/git.test.ts

```diff
@@ -0,0 +1,178 @@
+    // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+
+    it(`properly adds and rms a single file`, async () => {
+        const git = new Git({ localDir });
+        // console.log(`git initialized to ${git.localDir}`);
+        const respAdd = await git.add(`1970/0xxx/CVE-1970-0997.json`);
+        // console.log(`respAdd=${JSON.stringify(respAdd, null, 2)}`);
+        const status = await git.status();
+        // console.log(`status=${JSON.stringify(status, null, 2)}`);
+        expect(status.staged).toContain(`${localDir}/1970/0xxx/CVE-1970-0997.json`);
+        // expect(resp.modified).toContain(`${localDir}/1970/0xxx/CVE-1970-0002.json`);
+
+        // cleanup
+        const resp2 = await git.rm(`${process.cwd()}/${destDir}/CVE-1970-0997.json`);
+    });
+
+
+    it(`properly adds and rms multiple files`, async () => {
+        const git = new Git({ localDir });
+        // console.log(`process.cwd=${process.cwd()}`);
+        const respAdd = await git.add([
+            `${process.cwd()}/${destDir}/CVE-1970-0996.json`,
+            `${process.cwd()}/${destDir}/CVE-1970-0997.json`
+        ]);
+        // console.log(`respAdd=${JSON.stringify(respAdd, null, 2)}`);
+        const status = await git.status();
+        // console.log(`status=${JSON.stringify(status, null, 2)}`);
+        expect(status.staged).toContain(`${destDir}/CVE-1970-0996.json`);
+        expect(status.staged).toContain(`${destDir}/CVE-1970-0997.json`);
+
+        // cleanup
+        const resp2 = await git.rm([
+            `${process.cwd()}/${destDir}/CVE-1970-0996.json`,
+            `${process.cwd()}/${destDir}/CVE-1970-0997.json`
+        ]);
+    });
+
+
+    it(`logCommitHashInWindow() properly logs files`, async () => {
+        const git = new Git({ localDir });
+        const retval = await git.logCommitHashInWindow("2023-02-16T00:00:00.000Z", "2023-08-21T23:59:59.999Z");
+        // console.log(`retval=${JSON.stringify(retval, null, 2)}`);
+        expect(retval[0]).toEqual(`fa298ddd39500963b357823b802dba6952bc71d0`);
+        expect(retval[retval.length - 1]).toEqual(`207b9f2b82908afbd8d9d2270969f6781f9d39e4`);
+        expect(retval.length).toBe(26);
+    });
+
+
+    it(`logCommitHashInWindow() properly handles case when there are no commits`, async () => {
+        const git = new Git({ localDir });
+        const retval = await git.logCommitHashInWindow("1970-02-16T00:00:00.000Z", "1970-03-21T23:59:59.999Z");
+        expect(retval.length).toBe(0);
+    });
+
+
+    // @todo this test isn't working because the git history is different than in original development project
+    // it(`logChangedFilenamesInTimeWindow() properly logs files`, async () => {
+    //     const git = new Git({ localDir });
+    //     const retval = await git.logChangedFilenamesInTimeWindow("2023-09-07T21:40:46.000Z", "2023-09-07T21:42:34.000Z");
```

Comment:
can we use immutable historical data to test this? will need to find somewhere that solves this permanently

Reply:
#16